### PR TITLE
Fix own transactions in TxPool

### PR DESCRIPTION
**Pull request by LukaszRozmej**

### CODE REVIEW

File: src/Nethermind/Nethermind.TxPool/TxPool.cs [hunk 1]
```diff
@@ -51,7 +51,7 @@ public class TxPool : ITxPool, IDisposable
 
         private readonly object _locker = new();
 
-        private readonly ConcurrentDictionary<Address, AddressNonces> _nonces =
+        private readonly Dictionary<Address, AddressNonces> _nonces =
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.TxPool/TxPool.cs:54]: why change from concurrent to normal?
could we bring it back to Concurrent

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.TxPool/TxPool.cs:54]: Its always accessed under lock(), I don't think we need to make it concurrent then - please review.

File: src/Nethermind/Nethermind.TxPool/TxPool.cs [hunk 2]
```diff
@@ -518,25 +490,29 @@ public void Dispose()
...
+        private void Notify(PeerInfo peer, Transaction tx, bool isPriority)
         {
             try
             {
-                peer.SendNewTransaction(tx, isPriority);
-                Metrics.PendingTransactionsSent++;
-                if (_logger.IsTrace) _logger.Trace($"Notified {peer.Enode} about a transaction: {tx.Hash}");
+                if (!peer.NotifiedTransactions.Get(tx.Hash))
+                {
+                    peer.NotifiedTransactions.Set(tx.Hash);
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.TxPool/TxPool.cs:499]: better PeerInfo.SendTransaction and handle adding to cache internally (encapsulation)

File: src/Nethermind/Nethermind.TxPool/TxPool.cs [hunk 3]
```diff
@@ -518,25 +490,29 @@ public void Dispose()
...
+                    peer.Peer.SendNewTransaction(tx, isPriority);
+                    Metrics.PendingTransactionsSent++;
+                    if (_logger.IsTrace) _logger.Trace($"Notified {peer.Peer.Enode} about a transaction: {tx.Hash}");
+                }
             }
             catch (Exception e)
             {
-                if (_logger.IsError) _logger.Error($"Failed to notify {peer.Enode} about a transaction: {tx.Hash}", e);
+                if (_logger.IsError) _logger.Error($"Failed to notify {peer.Peer.Enode} about a transaction: {tx.Hash}", e);
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.TxPool/TxPool.cs:507]: PeerInfo.ToString

File: src/Nethermind/Nethermind.TxPool/TxPool.cs [hunk 4]
```diff
@@ -630,5 +604,33 @@ public void SetTransactionHash(Keccak transactionHash)
...
+            }
+
+            public PublicKey Id => Peer.Id;
+
+            public bool SendNewTransaction(Transaction tx, bool isPriority)
+            {
+                if (!NotifiedTransactions.Get(tx.Hash))
+                {
+                    NotifiedTransactions.Set(tx.Hash);
+                    Peer.SendNewTransaction(tx, isPriority);
```
Comment (MarekM25) [inline src/Nethermind/Nethermind.TxPool/TxPool.cs:626]: Shouldn't be `return Peer.SendNewTransaction(tx, isPriority); `?

**Review by MarekM25 — APPROVED** [review]

**Review by kasiaturbiarz — APPROVED** [review]

**Pull request merged**